starknet/docs/deploy.md

```diff
@@ -0,0 +1,146 @@
+- For mainnet: Ledger hardware wallet
+
+## Environment Setup
+
+```bash
+# Set your test environment
+export STARKNET_ENV=testnet
+
+# Set your test account (for online transactions)
+export STARKNET_PRIVATE_KEY=0x1234...
+export STARKNET_ACCOUNT_ADDRESS=0x5678...
+```
+
+## Command Options
+
+### Required Options
+- `--env`: Environment (testnet, mainnet)
+- `--contractConfigName`: Contract configuration name (must exist in config)
+- `--accountAddress`: Account address for deployment
+
+### Optional Options
+- `--constructorCalldata`: Constructor arguments as JSON array (default: '[]')
+- `--salt`: Salt for deterministic deployment address
+- `--privateKey`: Private key (required for online, not for offline)
+- `--offline`: Generate unsigned transaction file
+- `--estimate`: Estimate gas and display CLI arguments
+- `--nonce`: Account nonce (required for offline)
+- `--yes`: Skip confirmation prompts
+
+### Gas Options (Offline Only)
+- `--l1GasMaxAmount`: Maximum L1 gas amount
+- `--l1GasMaxPricePerUnit`: Maximum L1 gas price per unit
+- `--l2GasMaxAmount`: Maximum L2 gas amount
+- `--l2GasMaxPricePerUnit`: Maximum L2 gas price per unit
+- `--l1DataMaxAmount`: Maximum L1 data amount
+- `--l1DataMaxPricePerUnit`: Maximum L1 data price per unit
+
+## Deployment Workflow
+
+### For Testnet (Online)
+
+1. **Ensure contract is declared**:
+   ```bash
+   # Contract must be declared first
+   npx ts-node starknet/declare-contract.ts --contractConfigName MyContract ...
+   ```
+
+2. **Deploy the contract**:
+   ```bash
+   npx ts-node starknet/deploy-contract.ts \
+     --env testnet \
+     --contractConfigName MyContract \
+     --constructorCalldata '["0xparam1", "0xparam2"]' \
+     --salt 0x123 \
+     --privateKey $STARKNET_PRIVATE_KEY \
+     --accountAddress $STARKNET_ACCOUNT_ADDRESS
+   ```
+
+### For Mainnet (Offline)
+
+1. **Estimate gas on online machine**:
+   ```bash
+   npx ts-node starknet/deploy-contract.ts \
+     --env mainnet \
+     --contractConfigName MyContract \
+     --constructorCalldata '["0xparam1"]' \
+     --salt 0x123 \
+     --estimate \
+     --privateKey $STARKNET_PRIVATE_KEY \
+     --accountAddress $STARKNET_ACCOUNT_ADDRESS
+   ```
+
+2. **Copy gas parameters from output**:
+   ```
+   --l1GasMaxAmount 50000 --l1GasMaxPricePerUnit 100000000000 --l2GasMaxAmount 1000000 --l2GasMaxPricePerUnit 1000000000
+   ```
+
+3. **Generate unsigned transaction on offline machine**:
+   ```bash
+   npx ts-node starknet/deploy-contract.ts \
+     --env mainnet \
+     --contractConfigName MyContract \
+     --constructorCalldata '["0xparam1"]' \
+     --salt 0x123 \
+     --offline \
+     --nonce 5 \
+     --accountAddress $STARKNET_ACCOUNT_ADDRESS \
+     --l1GasMaxAmount 50000 \
+     --l1GasMaxPricePerUnit 100000000000 \
+     --l2GasMaxAmount 1000000 \
+     --l2GasMaxPricePerUnit 1000000000
+   ```
+
+4. **Follow offline signing workflow** (see main offline.md)
+
+## Constructor Calldata Format
```

Comment:
Ok, here's the answer to my question above =p

Is there a straightforward way to get this serialized? Could we perhaps add some utility scripts to do something similar to what starkli does?

Reply:
The same applies for governance and operators btw, because both contracts execute arbitrary contracts

Reply:
As we [discussed in slack](https://eig3r.slack.com/archives/C06A2QAPE2Z/p1752244212055769?thread_ts=1752219639.840129&cid=C06A2QAPE2Z), we'll leave this script for now.